src/core/license.rs

```diff
@@ -332,14 +338,33 @@ impl NewLicense {
         ];
 
         message.read(raw)?;
-        let encrypted_license_data =
-            cast!(DataType::Slice, message["EncryptedLicenseInfo"])?.to_vec();
+
+        let encrypted_license_data = cast!(
+            DataType::Slice,
+            cast!(DataType::Component, message["EncryptedLicenseInfo"])?["blobData"]
+        )?
+        .to_vec();
         let mac_data = cast!(DataType::Slice, message["MACData"])?.to_vec();
         Ok(Self {
             encrypted_license_data,
             mac_data,
         })
     }
+
+    // TODO(zmb3): remove this exception
+    #[allow(dead_code)]
```

Comment:
`expect` will trigger once the lint is no longer firing, so you can't forget to remove the exception (newly stabilized in rust 1.81.0!)
```suggestion
    #[expect(dead_code)]
```

Reply:
neat!